src/test/java/com/mongodb/jdbc/MongoSQLMock.java

```diff
@@ -284,4 +361,239 @@ MongoJsonSchemaResult generateSchema() {
         schemaResult.schema.mongoJsonSchema = generateMongoJsonSchema();
         return schemaResult;
     }
+
+    protected static MongoJsonSchema generateMongoJsonSchemaAllTypes() {
```

Comment:
I added a "collection" to the mock that contains all bson types. I added this `generateMongoJsonSchemaAllTypes` method and the `generateRowAllTypes` method so we could create a mock result set that has all types. I only use this "collection" for testing the `getString()` and `getObject().toString()` behavior since that is all that is really in scope for this ticket.